build.yaml

```diff
@@ -1832,14 +1832,19 @@ steps:
  - kind: runImage
    name: test_hailtop_batch
    image:
-     valueFrom: service_base_image.image
+     valueFrom: hail_run_image.image
    script: |
      set -ex
+     cd /io
+     tar xvf wheel-container.tar
+     python3 -m pip install --no-dependencies hail-*-py3-none-any.whl
```

Comment:
I do not want `hailtop.batch` to require a user to have the JVM installed (which `hl.hadoop_open` and `hl.hadoop_exists` require).

The tool we have available is `hailtop.google_storage.GCS`. GCS does not implement a file system API, but it can check existence and read `gs://` URLs.

Reply:
I don't understand. regenie.py needs to read both local and remote files.

Reply:
This also seems like a far afield problem given that we require users to install hail to get hailtop currently. https://hail.is/docs/batch/getting_started.html

Reply:
By normal filesystem operations, I mean `basedir`, `dirname`, and friends. AFAICT, the only operations you need are "does XXX exist" and "give me XXX as a string". Upon a second look, I realize that GCS hides all its synchronous operations as underscore-prefixed-names. Luckily, it's pretty simple to use the synchronous google cloud storage library directly:

```python3
from google.cloud import storage
from google.cloud.storage.blob import Blob
client = storage.Client()
blob = Blob.from_string("gs://danking/a", client)
if blob.exists():
    step = blob.download_as_string()
```

Reply:
I wrote a FS abstraction for Hail, and a concrete implementation GoogleCloudStorageFS last year. This does what I need in a 2 line change. To avoid transitive dependencies from requiring JVM, I moved two imports into the scope they're used in in FS.

Reply:
Re-reading your last comment, outside of this thread, I see you also don't want to import hail query in any way.